#! /bin/sh /usr/share/dpatch/dpatch-run
##  by  Ondrej Certik
##
## DP: demux for newer ffmpeg  0.svn20080206

@DPATCH@
Index: mplayer-1.0~rc2/libmpdemux/demux_lavf.c
===================================================================
--- mplayer-1.0~rc2.orig/libmpdemux/demux_lavf.c	2008-07-05 12:41:23.000000000 +0200
+++ mplayer-1.0~rc2/libmpdemux/demux_lavf.c	2008-07-05 12:41:36.000000000 +0200
@@ -61,7 +61,7 @@
 typedef struct lavf_priv_t{
     AVInputFormat *avif;
     AVFormatContext *avfc;
-    ByteIOContext pb;
+    ByteIOContext *pb;
     int audio_streams;
     int video_streams;
     int sub_streams;
@@ -288,9 +288,9 @@
     
     url_fopen(&priv->pb, mp_filename, URL_RDONLY);
     
-    ((URLContext*)(priv->pb.opaque))->priv_data= demuxer->stream;
+    ((URLContext*)(priv->pb->opaque))->priv_data= demuxer->stream;
         
-    if(av_open_input_stream(&avfc, &priv->pb, mp_filename, priv->avif, &ap)<0){
+    if(av_open_input_stream(&avfc, priv->pb, mp_filename, priv->avif, &ap)<0){
         mp_msg(MSGT_HEADER,MSGL_ERR,"LAVF_header: av_open_input_stream() failed\n");
         return NULL;
     }
Index: mplayer-1.0~rc2/libmpdemux/muxer_lavf.c
===================================================================
--- mplayer-1.0~rc2.orig/libmpdemux/muxer_lavf.c	2008-07-05 12:41:31.000000000 +0200
+++ mplayer-1.0~rc2/libmpdemux/muxer_lavf.c	2008-07-05 12:41:36.000000000 +0200
@@ -40,7 +40,7 @@
 typedef struct {
 	//AVInputFormat *avif;
 	AVFormatContext *oc;
-	ByteIOContext pb;
+	ByteIOContext *pb;
 	int audio_streams;
 	int video_streams;
 	int64_t last_pts;
@@ -317,7 +317,7 @@
 		av_freep(&(priv->oc->streams[i]));
 	}
 
-	url_fclose(&(priv->oc->pb));
+	url_fclose(priv->oc->pb);
 
 	av_free(priv->oc);
 }
